src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -18,19 +18,321 @@
+#endif
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+
+    case PumpOperationMode::kMinimum: {
+        uint8_t minLevel;
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kMinimum);
+        Attributes::EffectiveControlMode::Set(endpoint, PumpControlMode::kConstantSpeed);
+#ifdef EMBER_AF_PLUGIN_LEVEL_CONTROL
+        LevelControl::Attributes::MinLevel::Get(endpoint, &minLevel);
+        LevelControl::Attributes::CurrentLevel::Set(endpoint, minLevel);
+#endif
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+
+    case PumpOperationMode::kLocal: {
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kLocal);
+        Attributes::EffectiveControlMode::Set(endpoint, controlMode);
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+    }
+    Attributes::PumpStatus::Set(endpoint, pumpStatus);
+}
+
+CHIP_ERROR PumpConfigurationAndControlAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    emberAfDebugPrintln("Reading from PCC");
+
+    VerifyOrDie(aPath.mClusterId == PumpConfigurationAndControl::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PumpConfigurationAndControlAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    emberAfDebugPrintln("Writing to PCC");
+
+    VerifyOrDie(aPath.mClusterId == PumpConfigurationAndControl::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+} // namespace
+
+// SDK Callbacks
+
+template <typename T1, typename T2>
+bool IsFeatureSupported(EndpointId endpoint, EmberAfStatus (*getFn1)(chip::EndpointId endpointId, T1 & value),
+                        EmberAfStatus (*getFn2)(chip::EndpointId endpointId, T2 & value))
+{
+    EmberAfStatus status;
+
+    T1 value1;
+    T2 value2;
+
+    status = getFn1(endpoint, value1);
+    if (status == EMBER_ZCL_STATUS_SUCCESS)
+    {
+        if (!value1.IsNull())
+        {
+            status = getFn2(endpoint, value2);
+            if (status == EMBER_ZCL_STATUS_SUCCESS)
+            {
+                if (!value2.IsNull())
+                {
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
 
 void emberAfPumpConfigurationAndControlClusterServerInitCallback(EndpointId endpoint)
 {
+    bool constPressureSupported     = false;
+    bool constPropPressureSupported = false;
```

Comment:
I'm actually a bit confused which is the best name for this since a few different terms are used in the spec:

CompPressure, ProportionalPressure and Constant differential pressure.

Reply:
Yes, thats correct. I have also noticed that.